### PR TITLE
Fix potential Null pointer dereference in event-read-fifo.c

### PR DESCRIPTION
**Pull request by icy17**

### CODE REVIEW

File: sample/event-read-fifo.c [hunk 1]
```diff
@@ -139,6 +139,10 @@ main(int argc, char **argv)
 #else
 	/* catch SIGINT so that event.fifo can be cleaned up */
 	signal_int = evsignal_new(base, SIGINT, signal_cb, base);
+	if (signal_int == NULL) {
+		perror("evsignal_new");
```
Comment (azat) [inline sample/event-read-fifo.c:143]: Should have normal error path here (via goto, see other PRs comments)

**Review by azat — CHANGES_REQUESTED** [review]

File: sample/event-read-fifo.c (hunk 1 above)
Comment (icy17) [reply sample/event-read-fifo.c:143]: OK, I'll fix it

File: sample/event-read-fifo.c [hunk 2]
```diff
@@ -162,5 +161,16 @@ main(int argc, char **argv)
 #endif
 	libevent_global_shutdown();
 	return (0);
+
+err:
```
Comment (azat) [inline sample/event-read-fifo.c:165]: You forgot about `evfifo`

**Comment by azat** [comment]
Will be merged with https://github.com/libevent/libevent/pull/1673. Thank you!

**Pull request closed without merging**